Mage/MessageStream/MessageStream.cs

```diff
@@ -39,44 +39,51 @@ private static Logger Logger
 
 
 		//
-		private void InitializeMessageList()
+		private void Initialize()
 		{
+			// Stop the transport client if it exists
+			if (transportClient != null)
+			{
+				transportClient.Stop();
+			}
+
+			// Reset the session key
+			sessionKey = null;
+
+			// Initialize the message queue
 			currentMessageId = -1;
 			largestMessageId = -1;
 			messageQueue = new Dictionary<int, JToken>();
 			confirmIds = new List<string>();
 
-			Logger.Debug("Initialized message queue");
+			Logger.Debug("(Re)Initialized message queue");
```

Comment:
Would there be a way (or reason perhaps) to log an initial setup from an nth one?

Reply:
I'm not sure i understand this question